flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -450,6 +450,40 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "1111111111111111111111111111111111111111111111111111111111111111")
   }
 
+  @Test
+  def testRegexExtract(): Unit = {
```

Comment:
Can we add a test specifically doing the backslash escape case. For example: 
```
"foothebar".regexExtract("foo([\\w]+"), 1) // should return "thebar"
```

Reply:
Good point, here is a problem, I wrote this case to test  : 

```scala
testAllApis(
  "foothebar".regexExtract("foo([\\w]+)", 1),                 //OK,
  "'foothebar'.regexExtract('foo([\\\\w]+)', 1)",              //failed, got 'foo([\\w]+)' returns "", but if pass 'foo([\\w]+)' would get compile error.
  "REGEX_EXTRACT('foothebar', 'foo([\\\\w]+)', 1)",    //OK, but must pass four '\'
  "thebar"
)
```

It seems flink pre-process the regex which contains `\xxx`. A few days ago, we also met this issue when test `similar to` to match the regex which contains `\d`.

cc @twalthr

Reply:
This test case : 

```scala
    testSqlApi("REGEX_EXTRACT('foothebar', 'foo([\\\\w]+)', 1)", "thebar")
```

can pass